sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_visits_daily_v2/query.sql

```diff
@@ -0,0 +1,284 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_info.client_id AS client_id,
+    SAFE_CAST(
+      mozfun.norm.browser_version_info(client_info.app_display_version).major_version AS INT64
+    ) AS app_version,
+    normalized_os AS os,
+    normalized_channel AS channel,
+    client_info.locale AS locale,
+    normalized_country_code AS country,
+    metrics.string.newtab_homepage_category AS homepage_category,
+    metrics.string.newtab_newtab_category AS newtab_category,
+    metrics.boolean.pocket_enabled AS organic_content_enabled,
+    metrics.boolean.pocket_sponsored_stories_enabled AS sponsored_content_enabled,
+    metrics.boolean.topsites_sponsored_enabled AS sponsored_topsites_enabled,
+    metrics.boolean.topsites_enabled AS organic_topsites_enabled,
+    metrics.boolean.newtab_search_enabled AS newtab_search_enabled,
+    mozfun.newtab.is_default_ui_v1(
+      category,
+      name,
+      extra,
+      metrics.string.newtab_homepage_category,
+      metrics.string.newtab_newtab_category
+    ) AS is_default_ui,
+    category AS event_category,
+    name AS event_name,
+    extra AS event_details,
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    -- selecting the events we need to compute on
+    AND (
+      (category = 'newtab.search' AND name IN ('issued'))
+      OR (category = 'newtab.search.ad' AND name IN ('click'))
+      OR (
+        category = 'pocket'
+        AND name IN (
+          'click',
+          'impression',
+          'dismiss',
+          'thumb_voting_interaction',
+          'thumb_voting_interaction'
+        )
+      )
+      OR (category = 'topsites' AND name IN ('click', 'impression', 'dismiss'))
+      OR (
+        category = 'newtab'
+        AND name IN (
+          'opened',
+          -- weather
+          'weather_change_display',
+          'weather_open_provider_url',
+          'weather_location_selected',
+          -- wallpaper
+          'wallpaper_click',
+          'wallpaper_category_click',
+          'wallpaper_highlight_cta_click',
+          'wallpaper_highlight_dismissed',
+          -- topic_selection
+          'topic_selection_open',
+          'topic_selection_dismiss',
+          'topic_selection_topics_saved',
+          -- section
+          'sections_block_section',
+          'sections_follow_section',
+          'sections_unblock_section',
+          'sections_unfollow_section',
+          'inline_selection_click'
+        )
+      )
+    )
+)
+SELECT
+  submission_date,
+  client_id,
+  mozfun.map.get_key(event_details, 'newtab_visit_id') AS newtab_visit_id,
+  -- ANY_VALUE: because we do not expect those fields to differ within a visit
```

Comment:
Did you consider using ARRAY_AGG for consistent results and remove risks like
- incorrect aggregations between sources when there is more than one value
- inconsistent data after a backfill because a different value is selected
e.g. ARRAY_AGG(app_version ORDER BY submission_date DESC)[OFFSET(0)] AS app_version

Reply:
If there is a concern with values changing within the short duration of a visit, we could do something like picking the most recent value. But my thoughts are:
- the visit duration is very short and chances of different values showing is not common or may not be a concern
- `ANY_VALUE` would perform better

Reply:
I'd also expect ANY_VALUE to perform better. What's the average duration and the actual variation, let's say looking at 2024?

Reply:
The usage of `ANY_VALUE` has also been discussed with Nadia and Ilana as an acceptable way to apply for visit level dimensions. Jared also accepted because the current implementation `newtab_visits_v1` applies the same method.